site/source/docs/porting/pthreads.rst

```diff
@@ -39,15 +51,15 @@ The Emscripten implementation for the pthreads API should follow the POSIX stand
   USE_PTHREADS=1`` and the other with ``-s USE_PTHREADS=0``.
 
-- When the linker flag `-s PTHREAD_POOL_SIZE=<integer>` is not specified and `pthread_create()` is called, the new thread will not actually start to run immediately, but the main JS thread must yield execution back to browser first. This behavior is a result of `#1049079 <https://bugzilla.mozilla.org/show_bug.cgi?id=1049079>`_.
+- When the linker flag ``-s PTHREAD_POOL_SIZE=<integer>`` is not specified and ``pthread_create()`` is called, the new thread will not actually start to run immediately, but the main JS thread must yield execution back to browser first. This behavior is a result of `#1049079 <https://bugzilla.mozilla.org/show_bug.cgi?id=1049079>`_.
 
-- Currently several of the functions in the C runtime, such as filesystem functions like `fopen()`, `fread()`, `printf()`, `fprintf()` etc. are not multithreaded, but instead their execution is proxied over to the main application thread. Memory allocation via `malloc()` and `free()` is fully multithreaded though. This proxying can generate a deadlock in a special situation that native code running pthreads does not have. See `bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for more information and how to work around this until proxying is no longer needed in Emscripten.
+- Currently several of the functions in the C runtime, such as filesystem functions like ``fopen()``, ``fread()``, ``printf()``, ``fprintf()`` etc. are not multithreaded, but instead their execution is proxied over to the main application thread. Memory allocation via ``malloc()`` and ``free()`` is fully multithreaded though. This proxying can generate a deadlock in a special situation that native code running pthreads does not have. See `bug 3495 <https://github.com/emscripten-core/emscripten/issues/3495>`_ for more information and how to work around this until proxying is no longer needed in Emscripten.
 
-- In order to keep proxying as responsive as possible, whenever main thread calls to a function that performs a futex wait, e.g. `usleep()`, `emscripten_futex_wait()`, or `pthread_mutex_lock()`, the wait is done in very short time slices, which means that functions such as `usleep()` are not necessarily effective to conserve power. In order to save battery in the main thread, it is best to yield back to the browser runtime. When a pthread perform a futex wait, it sleeps in considerably longer slices.
+- In order to keep proxying as responsive as possible, whenever main thread calls to a function that performs a futex wait, e.g. ``usleep()``, ``emscripten_futex_wait()``, or ``pthread_mutex_lock()``, the wait is done in very short time slices, which means that functions such as ``usleep()`` are not necessarily effective to conserve power. In order to save battery in the main thread, it is best to yield back to the browser runtime. When a pthread perform a futex wait, it sleeps in considerably longer slices.
```

Comment:
We should probably go even further and say that these functions just busy-wait. I'm not really sure what the time slice has to do with anything?

Reply:
I agree.  This is one of the more confusing examples of language in the docs from the perspective of someone trying to reconcile the POSIX threading model with the web's.

Reply:
Yes, it does look like the code simply busy-waits. I'll remove the confusing stuff.

Reply:
Ah, I think this text was describing the browser's (Firefox's) behavior under the hood. Still very confusing, removing.